sled-diagnostics/src/logs.rs

```diff
@@ -0,0 +1,245 @@
+            let mut paths = Vec::new();
+            paths.push(entry.primary);
+            paths.extend(entry.debug);
+            paths.extend(entry.extra.into_iter().map(|e| e.1));
+            paths
+        })
+        .collect())
+}
+
+/// Find the relevant logs that should be returned as apart of a Support Bundle.
+pub fn find_logs() -> Result<SledDiagnosticsLogs, SledDiagnosticsLogError> {
+    let zones =
+        oxlog::Zones::load().map_err(SledDiagnosticsLogError::ZonePaths)?;
+
+    let mut result = SledDiagnosticsLogs { logs: BTreeMap::new() };
+    for zone in zones.zones.keys() {
+        let filter = oxlog::Filter {
+            current: true,
+            archived: true,
+            extra: true,
+            // avoid having to call `lstat` on each log file
+            show_empty: true,
+        };
+
+        let mut logs = Service::new();
+        for (svc, mut svclogs) in zones.zone_logs(&zone, filter) {
+            // Current
+            if let Some(current) = svclogs.current {
+                logs.insert(svc.clone(), vec![current.path]);
+            }
+
+            // Archived
+            logs.entry(svc.clone()).or_default().extend(
+                svclogs.archived.into_iter().rev().take(5).map(|s| s.path),
+            );
+
+            // Extra
+            let extra = std::mem::take(&mut svclogs.extra);
+            for (_, cockroach) in process_extra_logs(extra) {
+                logs.entry(svc.clone()).or_default().extend(cockroach.current);
+                logs.entry(svc.clone())
+                    .or_default()
+                    .extend(cockroach.rotated.into_iter().rev().take(5));
+            }
+        }
+        result.logs.insert(zone.clone(), logs);
+    }
+
+    Ok(result)
+}
+
+#[derive(Default, Debug, PartialEq)]
+struct CockroachExtraLog {
+    current: Vec<Utf8PathBuf>,
+    rotated: Vec<Utf8PathBuf>,
+}
+
+/// Walk over the extra logs for cockroachdb preserving order and sort them into
+/// the appropriate buckets.
+fn process_extra_logs(
+    extra: Vec<oxlog::LogFile>,
+) -> HashMap<String, CockroachExtraLog> {
+    // Known logging paths for cockroachdb:
+    // https://www.cockroachlabs.com/docs/stable/logging-overview#logging-destinations
+    let cockroach_logs = [
+        "cockroach",
+        "cockroach-health",
+        "cockroach-kv-distribution",
+        "cockroach-security",
+        "cockroach-sql-audit",
+        "cockroach-sql-auth",
+        "cockroach-sql-exec",
+        "cockroach-sql-slow",
+        "cockroach-sql-schema",
+        "cockroach-pebble",
+        "cockroach-telemetry",
+        // Not documented but found on our sleds
+        "cockroach-stderr",
+    ];
+
+    // Find all cockroachdb extra logs and split them up by file prefix
+    let mut sorted: HashMap<String, CockroachExtraLog> = HashMap::new();
+    for file in extra {
+        if let Some(file_name) = file.path.file_name() {
+            if let Some(file_parts) = file_name.split_once(".") {
```

Comment:
Why do we need to split the file_name like this? Can't we:

1. Check that the `file_name` starts with a prefix in `cockroach_logs`
2. Sort by the whole `file_name`? It's going to be sorted by prefix anyway

Reply:
I ended up doing it like this because the paths look like:
```
cockroach-health.log
cockroach-health.oxzcockroachdba3628a56-6f85-43b5-be50-71d8f0e04877.root.2025-01-31T21_43_26Z.011435.log
cockroach-health.oxzcockroachdba3628a56-6f85-43b5-be50-71d8f0e04877.root.2025-02-01T01_51_53Z.011486.log
...snip...
```
They get broken out into buckets so that we have the current generation and then all the rotated log files which should be preserving the order we find things in so that we can take the last n items.